File: backend/java/com/taskmanager/model/Task.java

package com.taskmanager.model;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import jakarta.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "tasks")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Task {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;

    @Column(nullable = false)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    private Priority priority = Priority.MEDIUM;

    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    private Status status = Status.TODO;

    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    private Category category = Category.WORK;

    @Column(nullable = false)
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'")
    private LocalDateTime createdAt = LocalDateTime.now();

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'")
    private LocalDateTime dueDate;

    public enum Priority {
        LOW, MEDIUM, HIGH
    }

    public enum Status {
        TODO, INPROGRESS, DONE
    }

    public enum Category {
        WORK, PERSONAL, STUDY, OTHER
    }
}

File: frontend/src/types/task.ts

export type Priority = 'low' | 'medium' | 'high';
export type Status = 'todo' | 'inprogress' | 'done';
export type Category = 'work' | 'personal' | 'study' | 'other';

export interface Task {
  id: string;
  title: string;
  description?: string;
  priority: Priority;
  status: Status;
  category: Category;
  createdAt: Date;
  dueDate?: Date;
}

// Backend response types
export interface TaskResponse {
  id: string;
  title: string;
  description?: string;
  priority: string;
  status: string;
  category: string;
  createdAt: string;
  dueDate?: string;
}

// For API requests
export interface CreateTaskRequest {
  title: string;
  description?: string;
  priority: Priority;
  status: Status;
  category: Category;
  dueDate?: string;
}

export interface UpdateTaskRequest extends CreateTaskRequest {
  id: string;
}
